packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -224,50 +274,91 @@ class SnackBar extends StatelessWidget {
     ];
     if (action != null) {
       children.add(ButtonTheme.bar(
-        padding: const EdgeInsets.symmetric(horizontal: _kSnackBarPadding),
+        padding: EdgeInsets.symmetric(horizontal: snackBarPadding),
         textTheme: ButtonTextTheme.accent,
         child: action,
       ));
     } else {
-      children.add(const SizedBox(width: _kSnackBarPadding));
+      children.add(SizedBox(width: snackBarPadding));
     }
-    final CurvedAnimation heightAnimation = CurvedAnimation(parent: animation, curve: _snackBarHeightCurve);
-    final CurvedAnimation fadeAnimation = CurvedAnimation(parent: animation, curve: _snackBarFadeCurve, reverseCurve: const Threshold(0.0));
-    Widget snackbar = SafeArea(
+    final CurvedAnimation heightAnimation =
+        CurvedAnimation(parent: animation, curve: _snackBarHeightCurve);
+    final CurvedAnimation fadeInAnimation =
+        CurvedAnimation(parent: animation, curve: _snackBarFadeInCurve);
+    final CurvedAnimation fadeOutAnimation = CurvedAnimation(
+      parent: animation,
+      curve: _snackBarFadeOutCurve,
+      reverseCurve: const Threshold(0.0),
+    );
+
+    final SnackBarBehavior snackBarBehavior = this.snackBarBehavior ?? snackBarTheme.snackBarBehavior ?? SnackBarBehavior.fixed;
+    final bool isFloatingSnackBar = snackBarBehavior == SnackBarBehavior.floating;
+
+    Widget snackBar = SafeArea(
       top: false,
+      bottom: !isFloatingSnackBar,
       child: Row(
         children: children,
         crossAxisAlignment: CrossAxisAlignment.center,
       ),
     );
-    snackbar = Semantics(
+
+    final double elevation = this.elevation ?? snackBarTheme.elevation ?? 6.0;
+    final Color backgroundColor = this.backgroundColor ?? snackBarTheme.backgroundColor ?? _kSnackBackground;
+    final ShapeBorder shape = this.shape
+        ?? snackBarTheme.shape
+        ?? (isFloatingSnackBar ? RoundedRectangleBorder(borderRadius: BorderRadius.circular(4.0)) : null);
+
+    snackBar = Material(
+      shape: shape,
+      elevation: elevation,
+      color: backgroundColor,
+      child: Theme(
+        data: darkTheme,
```

Comment:
How hard would it be to have the theme flip to lightTheme if the app were all dark themed?

Reply:
It would be doable, but I want to leave that out of the scope of this PR. Will handle later

Reply:
Ok! Put in a TODO with a ticket number.